warehouse/config/WHDefaultConfig.js

```diff
@@ -1,22 +1,17 @@
 const { coalesce } = require("./helpers");
 
 const rules = {
-  direct: {
-    messageId: "id",
-    anonymousId: "anonymous_id",
-    userId: "user_id",
-    request_ip: "context_request_ip",
-    "context.ip": "context_passed_ip",
-    sentAt: "sent_at",
-    timestamp: "timestamp",
-    receivedAt: "received_at",
-    originalTimestamp: "original_timestamp",
-    channel: "channel"
-  },
-
-  custom: {
-    context_ip: message => coalesce(message, ["context.ip", "request_ip"])
-  }
+  id: "messageId",
+  anonymous_id: "anonymousId",
+  user_id: "userId",
+  sent_at: "sentAt",
+  timestamp: "timestamp",
+  received_at: "receivedAt",
+  original_timestamp: "originalTimestamp",
+  channel: "channel",
+  context_ip: message => coalesce(message, ["context.ip", "request_ip"]),
```

Comment:
coalesce might not be the right name?
getFirstValidValue() or something that is easy to know the meaning from name.